src/objectExplorer/objectExplorerService.ts

```diff
@@ -144,25 +145,29 @@ export class ObjectExplorerService {
         return handler;
     }
 
-    public async expandNode(node: TreeNodeInfo, sessionId: string, promise: Deferred<TreeNodeInfo[]>): Promise<boolean> {
+    public async expandNode(node: TreeNodeInfo, sessionId: string, promise: Deferred<TreeNodeInfo[]>): Promise<boolean | undefined> {
         const expandParams: ExpandParams = {
             sessionId: sessionId,
             nodePath: node.nodePath
         };
         this._expandParamsToPromiseMap.set(expandParams, promise);
         const response = await this._connectionManager.client.sendRequest(ExpandRequest.type, expandParams);
-        if (!response) {
+        if (response) {
+            this._currentNode = node;
+            return response;
+        } else {
             this._expandParamsToPromiseMap.delete(expandParams);
+            promise.resolve(undefined);
+            return undefined;
```

Comment:
Not sure why you'd want to return undefined here. False should result in the same behavior and is just cleaner

Reply:
@Charles-Gagnon sqltoolsservice looks at the term 'databaseDisplayName` in the connection details options when listing out the object explorer nodes for it. This line specifically - https://github.com/microsoft/vscode-mssql/pull/1473/files/e000bbc9411d369a8e699c9fd650c036ba998573#diff-40da5394330734421e8030834675cd46R59

Reply:
Interesting, but it also seems to look at databaseName as well. Was that not working correctly before?

https://github.com/Microsoft/sqltoolsservice/blob/ec09ad21d291b33ac04b9e5ac4714ae958fbcfdb/src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/ObjectExplorerService.cs#L488

Reply:
Although I guess you're setting database, not databaseName. Is that maybe the issue here?

Reply:
Perhaps. I basically tried to see what's different in the extension and ADS and the `databaseDisplayName` was the only terms that wasn't being set on the extension side. Adding that made OE work as expected.

Reply:
Pretty odd. I'm not even sure what a "database display name is". We don't have display names for databases AFAIK - it's just the database name. I feel like something else is wrong here, but if this fixes the issue and there's no other problems seen in your testing I guess it's fine.